### ISSUE TITLE
Need to check for file permission before attempting to save.

### ISSUE THREAD

**Issue by carlosperate** [mu-editor/mu#207]
If we open a file that is read only for the user and try to save we don't safeguard for `PermissionError`:

```
Traceback (most recent call last):
  File "C:\workspace\git\microbit\mu_original\mu\logic.py", line 601, in save
    with open(tab.path, 'w', newline='') as f:
PermissionError: [Errno 13] Permission denied: 'C:/Users/xxx/mu_code/test_locked.py'
```
We should catch this, and then either:
a) Ask the user to manually remove the permission and try again
b) Ask if the user wants Mu to change the read permission and continue saving

**Comment by carlosperate** [comment]
@ntolls this one is still a bug, and now with the autosave it means that Mu crashes a few seconds after opening the read-only file:
```
Traceback (most recent call last):
  File "C:\workspace\mu-temp\mu\mu\logic.py", line 879, in autosave
    with open(tab.path, 'w', newline='') as f:
PermissionError: [Errno 13] Permission denied: 'C:/Users/carlo/mu_code/hello.py'
```

**Comment by tjguk** [comment]
I'll pick this up for now, since I'm working in the load/save space for issue #370. At least, please can others leave this area alone for the moment?

**Comment by carlosperate** [comment]
Thanks Tim!

**Comment by ntoll** [comment]
+1000 @tjguk

**Comment by ntoll** [comment]
Fixed by #399. I added a helpful message in a948debc2fe769df591effd97b1dd523f59beecc.
